README.md

```diff
@@ -1,118 +1,80 @@
+ 
 
-In structures where two or more folders are nested, any property defined for a folder will be overridden by any other property of the same name defined by one of its sub-folders.
+# About this Plugin
 
-## Develop
+The aim here is to remove the need to specify the same properties over
+and over again for all the jobs inside a folder.
 
-For ease of development I've added a `logging.properties` file that can be used by specifying it as an option to the
-Maven `hpi:run` target like so:
+In structures where two or more folders are nested, any property defined
+for a folder will be overridden by any other property of the same name
+defined by one of its sub-folders.
 
-    mvn hpi:run -Djava.util.logging.config.file=logging.properties
+## Configuring Folder Properties
 
-## Package
+Just create a
+[folder](https://plugins.jenkins.io/cloudbees-folder/),
+go to its configuration page and add as many properties as you need
+under the `Folder Properties` section.
 
-Since this is my first version of this and I'm still missing the tests...
+![](docs/images/folder-properties-config.png)
 
-    mvn package -DskipTests=true
+## Freestyle Jobs
 
-## Use
+Freestyle jobs must opt into the `Folder Properties` build wrapper from
+the `Build Environment` section of their configuration page in order to
+be able to access these properties as they would any other environment
+variable.
 
-In structures where two or more folders are nested, any property defined for a folder will be overridden by any other
-property of the same name defined by one of its sub-folders.
+![](docs/images/folder-properties-freestyle-config.png)
 
-<img src="./pics/folder-properties-1.png" align="center" alt="freestyle build env" style="width: 80%; height: auto"/>
+## Pipeline Jobs
 
-### Freestyle Jobs
+Pipeline jobs can use step `withFolderProperties` to access them:
 
-Freestyle jobs must opt into the `Folder Properties` build wrapper from the `Build Environment` section of their
-configuration page in order to be able to access these properties as they would any other environment variable.
+**Using folder properties in a pipeline job**
 
-<img src="./pics/freestyle-build-env.png" align="center" alt="freestyle build env" style="width: 80%; height: auto"/><br>
+``` syntaxhighlighter-pre
+withFolderProperties{
+    echo("Foo: ${env.FOO}")
+}
+```
 
-Only then will they inherit properties defined by their parent or ancestor folders —e.g. Running `echo $FOO` in a Shell build step:
+Jenkins deployments using some of the older versions of the [Structs
+Plugin](https://plugins.jenkins.io/structs/) will need
+to do this using the `wrap` meta-step:
 
-<img src="./pics/freestyle-example-1.png" align="center" alt="freestyle build env" style="width: 80%; height: auto"/><br>
+**Using folder properties in older pipeline jobs**
 
-#### SCM Step in Freestyle Jobs
+``` syntaxhighlighter-pre
```

Comment:
We usually use groovy syntax highlighter for Pipeline steps.
```suggestion
``` groovy
```

Reply:
fixed. coming on next commit.